### ISSUE TITLE
Time difference between reference frames for LISA inference

### ISSUE THREAD

**Issue by HumphreyWang** [gwastro/pycbc#4785]
When using the Gaussian noise model to run LISA inference, a time stamp difference showed in PE results if set `ref_frame = SSB` (in both injection and inference configs) instead of `ref_frame = LISA`. It indicates epochs in data and likelihood calculation might be inconsistent for LISA with SSB parameters

Take settings from #4686 but with `ref_frame` changed only, here's the result:
![lisa_smbhb_mass_tc](https://github.com/gwastro/pycbc/assets/68005354/8f7a0c40-f0d7-433b-bf97-eaf0a655415e)

**Comment by WuShichao** [comment]
I just tested it with `relbin` example https://pycbc.org/pycbc/latest/html/inference/examples/lisa_smbhb_inj_pe.html in SSB frame:
![image](https://github.com/gwastro/pycbc/assets/6574526/9024c917-720e-41fe-b0b3-936a01cc028f)
It works here.

**Comment by HumphreyWang** [comment]
@WuShichao You are right, it works for rebin model. BBHx will return an `Array` instead of a `FrequencySeries` if a frequency sequence is given. Then my suspicion is waveforms in rebin model do not need to keep epochs aligned (Array doesn't have this attribute), but in the Gaussian model they need to. So potential time inconsistency emerges for the Gaussian model.

I don't know if my description is misleading, maybe @spxiwh or @mj-will can help explain the issue.

**Comment by mj-will** [comment]
@WuShichao to add some more context to this. As it stands, BBHx always returns the waveform in the LISA reference frame if `sample_points` is not specified, see [here](https://github.com/gwastro/BBHX-waveform-model/blob/main/BBHX_Phenom.py#L354). This leads to a discrepancy between the waveform and `tc` [here](https://github.com/gwastro/pycbc/blob/master/pycbc/waveform/generator.py#L1189) which results in the waveform being shifted by an incorrect amount.

As @HumphreyWang mentioned, this doesn't apply to the relative binning case because it specifies `samples_points`.

Whilst we're observing this in BBHx this could occur with any waveform where `tc` is defined in a different frame to that of the waveform returned directly by the generator. It also therefore likely may not be a problem if the LISA response was being computed by PyCBC.
